client/driver/executor/executor_test.go

```diff
@@ -0,0 +1,158 @@
+package executor
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/nomad/client/allocdir"
+	"github.com/hashicorp/nomad/client/driver/env"
+	"github.com/hashicorp/nomad/nomad/mock"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/hashicorp/nomad/testutil"
+)
+
+var (
+	constraint = &structs.Resources{
+		CPU:      250,
+		MemoryMB: 256,
+		Networks: []*structs.NetworkResource{
+			&structs.NetworkResource{
+				MBits:        50,
+				DynamicPorts: []structs.Port{{Label: "http"}},
+			},
+		},
+	}
+)
+
+func mockAllocDir(t *testing.T) (string, *allocdir.AllocDir) {
+	alloc := mock.Alloc()
+	task := alloc.Job.TaskGroups[0].Tasks[0]
+
+	allocDir := allocdir.NewAllocDir(filepath.Join(os.TempDir(), alloc.ID))
+	if err := allocDir.Build([]*structs.Task{task}); err != nil {
+		log.Panicf("allocDir.Build() failed: %v", err)
+	}
+
+	return task.Name, allocDir
+}
+
+func testExecutorContext(t *testing.T) *ExecutorContext {
+	taskEnv := env.NewTaskEnvironment(mock.Node())
+	taskName, allocDir := mockAllocDir(t)
+	ctx := &ExecutorContext{
+		TaskEnv:       taskEnv,
+		TaskName:      taskName,
+		AllocDir:      allocDir,
+		TaskResources: constraint,
+	}
+	return ctx
+}
+
+func TestExecutor_Start_Invalid(t *testing.T) {
+	invalid := "/bin/foobar"
+	execCmd := ExecCommand{Cmd: invalid, Args: []string{"1"}}
+	ctx := testExecutorContext(t)
+	executor := NewExecutor(log.New(os.Stdout, "", log.LstdFlags))
+	_, err := executor.LaunchCmd(&execCmd, ctx)
+	if err == nil {
+		t.Fatalf("Expected error")
+	}
+	defer ctx.AllocDir.Destroy()
```

Comment:
This has to be at the top otherwise the fatal calls will make it so it never reaches this statement